### PR TITLE
LOS-225: Add periods to getStakeApy

### PR DESCRIPTION
**Pull request by j-tulforov**

### CODE REVIEW

File: packages/banner/src/components/LidoStakeBanner.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const LidoStakeBanner: React.FC<Props> = (props) => {
   const defiLink = getDefiLink(referrerId);
   const defiApy = getDefiApy();
 
-  const [apy, setApy] = useState(STATIC_DEFAULT_APY);
+  const [apy, setApy] = useState<string | number>(STATIC_DEFAULT_APY);
```
Comment (Drujininii) [inline packages/banner/src/components/LidoStakeBanner.tsx:40]: why string | number?

File: packages/common/stakeApy.ts [hunk 2]
```diff
@@ -1,22 +1,48 @@
 import { SOL_API_HOST } from './constants';
+import { ErrorWrapper } from '../core/src/utils/errorWrapper';
+import { ERROR_CODE } from '../core/src/constants/errors';
+
+export const STATIC_DEFAULT_APY = 9.35; // TODO think
```
Comment (Drujininii) [inline packages/common/stakeApy.ts:5]: maybe dash or hyphen? "–"

File: packages/common/stakeApy.ts [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
+    twoWeeks: ApyData;
+    oneMonth: ApyData;
+    threeMonth: ApyData;
+    sinceLaunch: ApyData;
   };
 };
 
-export const STATIC_DEFAULT_APY = '5.74'; // TODO think
+export const getMaxApy = (data: StakeApyResponse['data']) => {
+  return Object.values(data).reduce((max, curr) => {
```
Comment (Drujininii) [inline packages/common/stakeApy.ts:29]:
```suggestion
  return Math.max(...Object.values(data))
```

File: packages/common/stakeApy.ts [hunk 4]
```diff
@@ -1,22 +1,48 @@
...
+export const getStakeApy = async (): Promise<StakeApyResponse['data'] & { max: ApyData }> => {
+  return fetch(`${SOL_API_HOST}/v1/apy/complete`, { mode: 'cors' })
+    .then<StakeApyResponse>((res) => res.json())
+    .then(({ data }) => ({
+      max: getMaxApy(data),
+      ...data,
+    }))
+    .catch(() => {
+      throw new ErrorWrapper({ code: NO_APY_DATA });
+    });
```
Comment (Drujininii) [inline packages/common/stakeApy.ts:47]: 🪐

**Review by Drujininii — APPROVED** [review]

File: packages/common/stakeApy.ts (hunk 3 above)
Comment (j-tulforov) [reply packages/common/stakeApy.ts:29]: image
 please look at all function. I need to get link to max apy key, not value. If I need only value, I would write like this:
 
 ```ts
 Math.max(...Object.values(data).map(({apy}) => apy));
 ```

File: packages/banner/src/components/LidoStakeBanner.tsx (hunk 1 above)
Comment (j-tulforov) [reply packages/banner/src/components/LidoStakeBanner.tsx:40]: because of max.apy.**toFixed(2)** (toFixed returns string)

of course, we can add +, or `toNumber()`. But it's not necessary, because React will convert it to string again during rendering. So it's just typings, not requirement to make them similar.

File: packages/common/stakeApy.ts [hunk 5]
```diff
@@ -1,22 +1,48 @@
...
-  }
+export const getStakeApy = async (): Promise<StakeApyResponse['data'] & { max: ApyData }> => {
+  return fetch(`${SOL_API_HOST}/v1/apy/complete`, { mode: 'cors' })
+    .then<StakeApyResponse>((res) => res.json())
+    .then(({ data }) => ({
+      max: getMaxApy(data),
+      ...data,
+    }))
+    .catch(() => {
+      throw new ErrorWrapper({ code: NO_APY_DATA });
```
Comment (Tataklim) [inline packages/common/stakeApy.ts:46]: Why don't you want add error field here?

```
.catch((error) => {
      throw new ErrorWrapper({ error, code: NO_APY_DATA });
}
```

File: packages/core/src/utils/errorWrapper.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { ERROR_CODE, ERROR_CODE_DESC, ERROR_MESSAGE } from '@/constants';
+import { ERROR_CODE, ERROR_CODE_DESC, ERROR_MESSAGE } from '../constants/errors';
```
Comment (Tataklim) [inline packages/core/src/utils/errorWrapper.ts:1]: `@constants/errors`?

File: packages/core/src/statistics/lidoStatistics.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ export async function getLidoStatistics(this: SolidoSDK) {
   ]);
 
   return {
-    apy,
+    stakeApy: stakeApy.max,
```
Comment (Tataklim) [inline packages/core/src/statistics/lidoStatistics.ts:15]: Why `stakeApy`? I understand that it would be a major versions
But is this change really necessary?

Comment (Tataklim) [inline packages/core/src/statistics/lidoStatistics.ts:15]: Why `stakeApy`? I understand that it would be a major versions
But is this change really necessary?

File: packages/banner/src/components/LidoStakeBanner.tsx (hunk 1 above)
Comment (Drujininii) [reply packages/banner/src/components/LidoStakeBanner.tsx:40]: In some case it would be a bug in future. Better use +max.apy.toFixed(2) and strict number typing

Comment (j-tulforov) [reply packages/banner/src/components/LidoStakeBanner.tsx:40]: I don't think so, it's too primitive case, both types are ok. But okay, let it be.

File: packages/common/stakeApy.ts (hunk 5 above)
Comment (j-tulforov) [reply packages/common/stakeApy.ts:46]: oh yeah, thanks

File: packages/core/src/statistics/lidoStatistics.ts (hunk 7 above)
Comment (j-tulforov) [reply packages/core/src/statistics/lidoStatistics.ts:15]: because apy field now have this type:
```ts
type ApyData = {
  apy: number;
  apr: number;
  intervalPrices: {
    beginEpoch: number;
    endEpoch: number;
  };
};
```

So, it's weird to see code like this: `apy.apy`, or `apy.apr`. If you have a better idea, go.

File: packages/core/src/utils/errorWrapper.ts (hunk 6 above)
Comment (j-tulforov) [reply packages/core/src/utils/errorWrapper.ts:1]: Unfortunately can't use, because this file is going to banner bundle too.

But only right now I thought that it would be better to put it in common dir, do u agree?

Comment (Tataklim) [reply packages/core/src/utils/errorWrapper.ts:1]: Yeah, let's do it

File: packages/core/src/statistics/lidoStatistics.ts (hunk 7 above)
Comment (j-tulforov) [reply packages/core/src/statistics/lidoStatistics.ts:15]: Or maybe even return not all structure, just like before, return only max value?

Comment (Tataklim) [reply packages/core/src/statistics/lidoStatistics.ts:15]: Oh, u ar right, got it
maybe something like `yield`?

Comment (Tataklim) [reply packages/core/src/statistics/lidoStatistics.ts:15]: But I think we can also return one value, cause clints could orient on values in intervalPrices that could differ

File: packages/banner/src/components/LidoStakeBanner.tsx (hunk 1 above)
Comment (j-tulforov) [reply packages/banner/src/components/LidoStakeBanner.tsx]: done

File: packages/common/stakeApy.ts (hunk 5 above)
Comment (j-tulforov) [reply packages/common/stakeApy.ts]: done

File: packages/core/src/utils/errorWrapper.ts (hunk 6 above)
Comment (j-tulforov) [reply packages/core/src/utils/errorWrapper.ts]: done

File: packages/core/src/statistics/lidoStatistics.ts (hunk 7 above)
Comment (j-tulforov) [reply packages/core/src/statistics/lidoStatistics.ts]: done

**Review by Tataklim — APPROVED** [review]
